### PR TITLE
resolve: Update glob bindings in case of ambiguities (non-recursively)

### PR DESCRIPTION
**Pull request by petrochenkov**
One independent subset of https://github.com/rust-lang/rust/pull/112743 for a mini crater run.
cc @bvanjoi 

TODO: `@craterbot check p=1 crates=https://crater-reports.s3.amazonaws.com/pr-112743/retry-regressed-list.txt`

### CODE REVIEW

File: compiler/rustc_resolve/src/imports.rs [hunk 1]
```diff
@@ -324,6 +324,8 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                         } else if !old_binding.vis.is_at_least(binding.vis, this.tcx) {
                             // We are glob-importing the same item but with greater visibility.
                             resolution.binding = Some(binding);
+                        } else if binding.is_ambiguity() {
+                            resolution.binding = Some(binding);
```
Comment (bvanjoi) [inline compiler/rustc_resolve/src/imports.rs:328]: Can it solve #112713? I'm not sure.

**Comment by petrochenkov** [comment]
@craterbot check p=1 crates=https://crater-reports.s3.amazonaws.com/pr-112743/retry-regressed-list.txt

**Comment by petrochenkov** [comment]
@craterbot check p=1 crates=https://gist.githubusercontent.com/petrochenkov/78c5487fedf991fad040ff42a8a36e69/raw/cc4e227ef0fe8d65cc5d107ed84847d7b6bd5a94/gistfile1.txt

**Pull request closed without merging**